src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningKeyMapper.java

```diff
@@ -26,30 +28,33 @@
 
 class HashPartitioningKeyMapper {
 
-    @VisibleForTesting
-    static final String DELIMITER = ".";
-
-    private static final String DELIMITER_PATTERN = "\\.";
-
     static class HashPartitioningKeyPrefixFunction {
 
+        // see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-partition-sort-keys
+        private static final int MAX_HASH_KEY_LENGTH = 2048;
+
```

Comment:
As discussed in the previous PR, using length-based encoding so that we can easily use non-string contexts in the future if we want.